File: travelcard-function/Models/TravelcardModel.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace TravelcardFunction.Models
{
    public class TravelcardDto
    {
        public int Id { get; set; }
        public TravelcardType TravelcardType { get; set; }
        public DateTime TravelcardValidFrom { get; set; }
        public DateTime TravelcardValidTo { get; set; }
        public string? TravelcardName { get; set; }
        public string? TravelcardNumber { get; set; }
        public DateTime TravelcardRequestedDate { get; set; }
        public string TravelcardTransactionReference { get; set; } = string.Empty;
        public DateTime? TravelcardUsableTo { get; set; }
        public IEnumerable<CardholderDto> Cardholders { get; set; } = Array.Empty<CardholderDto>();
    }

    public class CardholderDto
    {
        public int Id { get; set; }
        public int TravelcardId { get; set; }
        public string CardholderTitle { get; set; } = string.Empty;
        public string CardholderForename { get; set; } = string.Empty;
        public string CardholderSurname { get; set; } = string.Empty;
        public CardholderType CardholderType { get; set; }
        public string CardholderPhotoName { get; set; } = string.Empty;
        public string? CardholderPhotoRRSKey { get; set; }
        public string? CardholderPhotoURL { get; set; }
        public string? CardholderPhotoKey { get; set; }
    }
}

File: app/models/travelcard_model.py
from dataclasses import dataclass
from datetime import datetime


@dataclass(slots=True)
class TravelCardData:
    travelcard_type: str
    travelcard_valid_from: datetime
    travelcard_valid_to: datetime
    travelcard_name: str | None
    travelcard_number: str
    travelcard_requested_date: datetime
    travelcard_transaction_reference: str
    travelcard_usable_to: datetime | None


@dataclass(slots=True)
class CardholderData:
    travelcard_id: int
    cardholder_title: str
    cardholder_forename: str
    cardholder_surname: str
    cardholder_type: str
    cardholder_photo_name: str
    cardholder_photo_rrs_key: str | None
    cardholder_photo_url: str | None
    cardholder_photo_key: str | None